lz4.go

```diff
@@ -154,12 +154,12 @@ func (w *Writer) writeFrame(src []byte) (int, error) {
 
 func (w *Writer) nextInputBuffer() []byte {
 	w.inpBufIndex = (w.inpBufIndex + 1) % 2
-	tmpSlice := reflect.SliceHeader{
-		Data: uintptr(w.compressionBuffer[w.inpBufIndex]),
-		Len:  streamingBlockSize,
-		Cap:  streamingBlockSize,
-	}
-	return *(*[]byte)(unsafe.Pointer(&tmpSlice))
+	var result []byte
+	resultHeader := (*reflect.SliceHeader)(unsafe.Pointer(&result))
+	resultHeader.Data = uintptr(w.compressionBuffer[w.inpBufIndex])
+	resultHeader.Len = streamingBlockSize
+	resultHeader.Cap = streamingBlockSize
+	return result
```

Comment:
We should use [`unsafe.Slice`](https://pkg.go.dev/unsafe#Slice) now that it's available to us

Reply:
I totally forgot about this, thanks! This is much better. This now requires Go 1.17, but I think that is okay also. Fixed!